R/commit-version.R

```diff
@@ -3,41 +3,46 @@ commit_version_impl <- function() {
 
   if (is_last_commit_bump()) {
     cli_alert("Resetting to previous commit.")
-    git2r::reset(git2r::revparse_single(revision = "HEAD^"))
+    gert::git_reset_soft(gert::git_log(max = 2)[2, "commit"])
     amending <- TRUE
   } else {
     amending <- FALSE
   }
 
-  git2r::add(".", c("DESCRIPTION", news_path))
-  if (length(git2r::status(".", unstaged = FALSE, untracked = FALSE)$staged) > 0) {
+  # need to get paths from git status to account for subdirs
+  news_path <- grep("DESCRIPTION", gert::git_status()$file, value = TRUE)
+  descr_path <- grep("NEWS.md", gert::git_status()$file, value = TRUE)
+
+  gert::git_add(c(descr_path, news_path))
+
+  if (length(gert::git_status(staged = TRUE)$staged) > 0) {
     cli_alert("Committing changes.")
-    git2r::commit(".", get_commit_message())
+    gert::git_commit(get_commit_message(), repo = gert::git_info()$path)
   }
 
   amending
 }
 
 is_last_commit_bump <- function() {
-  git2r::last_commit()$message == get_commit_message()
+  gert::git_log(max = 1)$message == get_commit_message()
 }
 
 get_commit_message <- function(version) {
   desc <- desc::desc(file = "DESCRIPTION")
   version <- desc$get_version()
 
-  paste0("Bump version to ", version)
+  paste0("Bump version to ", version, "\n")
```

Comment:
What's the purpose of the final newline? Does `git_log()` have one?

Reply:
readbility

> Does git_log() have one?

no